docs/proving-authentication.mdx

```diff
@@ -0,0 +1,189 @@
+first step is to plan at which phase during your application's authentication
+process will you want to use it.
+
+Wallets will optionally return to your application the `account-proof` service
+after completing wallet authentication. The suggested order of operations of how
+your application might use `account-proof` begins with:
+
+- Optional: You application may optionally configure FCL with an application
+  specific Domain Tag. A Domain Tag helps scope the cryptographic proof
+  generated for the `account-proof` service to your application. Your
+  application can configure a Domain Tag like:
+
+```javascript
+import {config} from "@onflow/fcl"
+
+config({
+    "fcl.domainTag": "YourApplicationsDomainTag"
+})
+```
+
+- Your application calls `fcl.authenticate()` to begin the user authentication
+  process.
+- Your user's wallet will optionally return the `account-proof` service as part
+  of authentication.
+- Your application extracts the data within the `account-proof` service.
+
+The data within the `account-proof` service will look like:
+
+```javascript
+{
+  f_type: "Service",  // Its a service!
+  f_vsn: "1.0.0",  // Follows the v1.0.0 spec for the service
+  type: "account-proof",  // the type of service it is
+  method: "DATA",  // Its data!
+  uid: "amazing-wallet#account-proof",  // A unique identifier for the service            
+  data: {
+    f_type: "account-proof",
+    f_vsn: "1.0.0"
+    signatures: [CompositeSignture],
+    address: "0xUSER",  // The users address
+    message: "abc123def456", // A random message
+    timestamp: 1630705495551, // UNIX timestamp
+    domainTag: "AWESOME_DAPP", // Optional, defaults to user signature domain tag `FLOW-V0.0-user`
+  }         
+}
+```
+
+- Your application will send the `account-proof` service data structure to it's
+  backend. 
+
+Once your application receives the `account-proof` data structure, it can then
+begin to use it to prove user authentication. The steps that your application's
+backend must follow in using this data are specific.
+
+- First your application must determine wether a user with the `address`
+  provided in the `account-proof` data structure has authenticated with your
+  application before.
+    - If a user with the `address` has authenticated with your application
+      before, then your system must query the `timestamp` associated with the
+      previous authentication for this user.
+        - If the `timestamp` is less than or equal to the `timestamp` associated
+          with the users previous authentication, your system must abort and
+          reject the authentication attempt.
+        - If the `timestamp` is greater than the `timestamp` associated with the
+          users previous authentication, then continue.
+    - If a user with the `address` _has not_ authenticated with your application
+      before, then continue.
+
+- Your application must determine whether the `CompositeSignture` in the
```

Comment:
Can the application not just confirm this on Flow by using`verifyUSerSignatures` with account key and domainTag?
Is reconstruction actually necessary? 🤔

Reply:
I believe so! The Data that gets reconstructed in this step becomes whats passed into `fcl.verifyUserSignature` as it's first parameter, the "message"

Reply:
this should be a WalletUtil helper...

Reply:
yeah @orodio agreed, saw this need while working on dev-wallet integration.

Reply:
Added WalletUtils for this https://github.com/onflow/fcl-js/pull/801/files